.github/workflows/build.yaml

```diff
@@ -7,6 +7,9 @@ on:
   workflow_dispatch:
     branches: [main]
 
+permissions:
+  contents: write
+
```

Comment:
Could you please set those permissions only on publish-doc job? The build job don't need those permissions, right?